src/TesApi.Web/Storage/StorageAccessProvider.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace TesApi.Web.Storage;
+
+/// <summary>
+/// 
+/// </summary>
+public abstract class StorageAccessProvider : IStorageAccessProvider
+{
+    /// <summary>
+    /// Cromwell path prefix
+    /// </summary>
+    protected const string CromwellPathPrefix = "/cromwell-executions/";
+    /// <summary>
+    /// Executions path prefix
+    /// </summary>
+    protected const string BatchPathPrefix = "/executions/";
+
+    /// <summary>
+    /// Logger instance. 
+    /// </summary>
+    protected readonly ILogger logger;
+    /// <summary>
+    /// Azure proxy instance.
+    /// </summary>
+    protected readonly IAzureProxy azureProxy;
+
+    /// <summary>
+    /// Provides base methods for blob storage access and local input mapping.
+    /// </summary>
+    /// <param name="logger">Logger <see cref="ILogger"/></param>
+    /// <param name="azureProxy">Azure proxy <see cref="IAzureProxy"/></param>
+    public StorageAccessProvider(ILogger logger, IAzureProxy azureProxy)
+    {
+        this.logger = logger;
+        this.azureProxy = azureProxy;
+
+    }
+
+    /// <inheritdoc />
+    public async Task<string> DownloadBlobAsync(string blobRelativePath)
+    {
+        try
+        {
+            return await this.azureProxy.DownloadBlobAsync(new Uri(await MapLocalPathToSasUrlAsync(blobRelativePath)));
+        }
+        catch
+        {
+            return null;
```

Comment:
TODO throw?

Reply:
Yes, agree. I will add an issue for this as this was the existing flow.